src/Storages/StorageMerge.cpp

```diff
@@ -1668,6 +1672,44 @@ std::optional<UInt64> StorageMerge::totalRowsOrBytes(F && func) const
     return first_table ? std::nullopt : std::make_optional(total_rows_or_bytes);
 }
 
+void StorageMerge::setTableToWrite(
+    const std::optional<String> & table_to_write_,
+    const String & source_database_name_or_regexp_,
+    bool database_is_regexp_)
+{
+    if (!table_to_write_.has_value())
+    {
+        table_to_write = std::nullopt;
+        return;
+    }
+
+    auto qualified_name = QualifiedTableName::parseFromString(*table_to_write_);
+
+    if (qualified_name.database.empty())
+    {
+        if (database_is_regexp_)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Argument 'table_to_write' must contain database if 'db_name' is regular expression.");
+
+        qualified_name.database = source_database_name_or_regexp_;
+    }
+
+    table_to_write = qualified_name;
+}
+
+SinkToStoragePtr StorageMerge::write(
+    const ASTPtr & query,
+    const StorageMetadataPtr & metadata_snapshot,
+    ContextPtr context_,
+    bool async_insert)
+{
+    if (!table_to_write.has_value())
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Method write is not allowed in storage {} without described table to write.", getName());
+
+    auto database = DatabaseCatalog::instance().getDatabase(table_to_write->database);
+    auto table = database->getTable(table_to_write->table, context_);
+    return table->write(query, metadata_snapshot, context_, async_insert);
```

Comment:
What happens if there is no table with given name?

Reply:
Error "Table ... does not exists". The same as in attempt to write directly in that non-existed table.
```
CREATE TABLE test03373_db.test03373_merge_wr_1 (key UInt32, value UInt32) ENGINE=Merge(test03373_db, 'test03373_table_\d+', test03373_table_2);

INSERT INTO test03373_db.test03373_merge_wr_1 VALUES (1,1)

Code: 60. DB::Exception: Received from localhost:9000. DB::Exception: Table test03373_db.test03373_table_2 does not exist. Maybe you meant test03373_db.test03373_merge_ro?. (UNKNOWN_TABLE)


INSERT INTO test03373_db.test03373_table_2 VALUES (1,1)

Code: 60. DB::Exception: Received from localhost:9000. DB::Exception: Table test03373_db.test03373_table_2 does not exist. Maybe you meant test03373_db.test03373_merge_ro?. (UNKNOWN_TABLE)
```